### ISSUE TITLE
cpp-coveralls 0.4.0 came and broke Travis build

### ISSUE THREAD

**Issue by pfalcon** [micropython/micropython#3246]
Last unfailed build:

https://travis-ci.org/micropython/micropython/builds/257565772

~~~
Successfully installed cpp-coveralls-0.3.12
~~~

First failing build:

https://travis-ci.org/micropython/micropython/builds/258107900

~~~
Successfully installed cpp-coveralls-0.4.0
...
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 863, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py", line 2577, in requires
    "%s has no such extra feature %r" % (self, ext)
pkg_resources.UnknownExtra: urllib3 1.7.1 has no such extra feature 'secure'
The command "(cd unix && coveralls --root .. --build-root . --gcov $(which gcov) --gcov-options '\-o build-coverage/' --include py --include extmod)" exited with 1.
~~~

Proposed solution: force-install cpp-coveralls==0.3.12

**Comment by pfalcon** [comment]
> Proposed solution: force-install cpp-coveralls==0.3.12

I implemented this workaround in f578947ae3fee5610c5bc1123baf878b92eaa248, because I'd like to perform refactors and have CI test them. Keeping this open if someone wants to dig deeper.

**Comment by dpgeorge** [comment]
The fix is discussed in https://github.com/eddyxu/cpp-coveralls/issues/118 and #3306 provides a working implementation.

**Comment by dpgeorge** [comment]
Fixed by #3306.
